### PR TITLE
Update devcontainer to Rust 1.80.1

### PR DESCRIPTION
**Pull request by windsource**
For #348 we need to update to Rust 1.80.1. The previous base container image `mcr.microsoft.com/devcontainers/rust` does not provide that Rust version. For that reason I have also changed the base image version to `ubuntu:24.04` and used `rustup` to install Rust.

Issues: #89

# Definition of Done

The PR shall be merged only if all items mentioned in [CONTRIBUTING.md](https://github.com/eclipse-ankaios/ankaios/blob/main/CONTRIBUTING.md#how-to-contribute) have been followed. In case an item is not applicable as described, please provide a short explanation in the description.

- [ ] All steps in [CONTRIBUTING.md](https://github.com/eclipse-ankaios/ankaios/blob/main/CONTRIBUTING.md#how-to-contribute) have been handled

### CODE REVIEW

**Comment by inf17101** [comment]
I have tested the ubuntu version with the following:

- debug build
- aarch64 cross build
- running all system tests
- serving the user documentation
- generate test coverage
- running an control interface example

No findings. Runs smooth.

File: .devcontainer/Dockerfile.base [hunk 1]
```diff
@@ -12,13 +12,28 @@
...
+        apt install -y gcc-aarch64-linux-gnu; \
+    elif [ "$TARGETARCH" = "arm64" ]; then \
+        apt install -y gcc-x86-64-linux-gnu; \
+    fi \
+    && apt -y install \
+    # Basics
+    git \
+    zsh \
+    curl \
+    iputils-ping \
```
Comment (inf17101) [inline .devcontainer/Dockerfile.base:35]: Do we really need this package?

Comment (windsource) [reply .devcontainer/Dockerfile.base:35]: You mean `iputils-ping`? Me and @krucod3 were missing that in the previous dev container. It provides the ping command which is required to test various connections problems. The size of the corresponding .deb file is only 43 kB so it should be OK.

**Comment by windsource** [comment]
Comparing the sizes:

| Image | Base image | Size for amd64 |
| ----- | ---------- | --------------- |
| ghcr.io/eclipse-ankaios/devcontainer-base:0.9.2 | mcr.microsoft.com/devcontainers/rust:1-1-bookworm | 4.28 GB |
| ghcr.io/eclipse-ankaios/devcontainer-base:0.10.0 | ubuntu:24.04 | 3.66 GB |

**Comment by inf17101** [comment]
I have also tried it out on another platform Mac M1 and it works.

**Comment by inf17101** [comment]
I suggest to add the following lines to `.cargo/config.toml`:

```rust
[target.x86_64-unknown-linux-musl]
linker="x86_64-linux-gnu-gcc"
```

Then I am able to build the default target x86... on Mac M1 with the default debug build command `cargo build`.

**Comment by inf17101** [comment]
I have tested in addition on Mac M1:

- coverage report
- building the user documenation

It works.

**Review by inf17101 — APPROVED** [review]
LGTM👍

**Pull request merged**